### DISCUSSION TITLE
How to get device_id for sso login

### DISCUSSION THREAD

**Discussion by ankitvermaAT** [tauri-apps/tauri#10863] [Q&A] [answered]
Hi, I’m developing a Windows app using Tauri with React.js. I need to implement SSO login and require a device ID that is consistently unique for a specific user each time they use the app on the same system.
Could you please suggest an approach that ensures no issues occur during the app upload to the store.

**Comment by charlesschaefer** [comment]
Hi @ankitvermaAT, maybe you could use the mac address of the network interface?

**Reply by ankitvermaAT** [reply, marked as answer]
Hi @charlesschaefer, Thanks for your valuable response.
But for using mac address we got below concerns...

**Privacy Concerns:** because using the MAC address as a unique identifier raises privacy concerns, as it is a persistent identifier that could be used to track users across different networks.

**Platform Limitations:** On some platforms, access to the MAC address may be restricted due to privacy policies or security settings. For example, modern versions of iOS and Android have limited or obfuscated access to MAC addresses.

**Security Risks:** MAC addresses are generally considered sensitive information, and exposing or storing them can introduce security risks.

can we use UUID :
#[tauri::command]
fn generate_device_id() -> String {
    use std::process::Command;
    let output = Command::new("wmic")
        .arg("csproduct")
        .arg("get")
        .arg("uuid")
        .output()
        .expect("Failed to execute command");
    let uuid = String::from_utf8_lossy(&output.stdout).trim().to_string();
    uuid
}
